BloodyMary/Navigation/Router.swift

```diff
@@ -0,0 +1,281 @@
+//
+//  BloodyMary
+//
+//  Copyright © TheInkedEngineer. All rights reserved.
+// 
+
+import UIKit
+
+/// The `Router` is the main point  that manages showing and hiding views.
+/// When instantiated it requires a configuration that provides it with the various routes and destinations.
+public struct Router {
+  
+  // MARK: - Properties
+  
+  /// Possibile errors while performing the routing operations.
+  public enum RoutingError {
+    /// ViewController missing from `screensAndDestinations`.
+    case viewControllerNotFound
+    /// Failed to properly cast the viewmodel.
+    case failedToAssignViewModel
+    
+    /// The message to display associated to the error.
+    var message: String {
+      switch self {
+      case .viewControllerNotFound:
+        return "Identifier is not associated to a RoutableViewController. Update your `screensAndDestinations`."
+        
+      case .failedToAssignViewModel:
+        return "Could not assign the viewModel properly to your vc."
+      }
+    }
+  }
+  
+  /// The queue on which all navigation is done.
+  private let routingQueue = DispatchQueue(label: "routing queue")
+  
+  /// The dictionary containing the routes and their destination view controllers.
+  private let screensAndDestinations: [ScreenIdentifier: RoutableViewController.Type]
+  
+  /// Initiates a `Router` and returns it with the proper configuration.
+  public init(with configuration: RoutingConfigurationProvider) {
+    self.screensAndDestinations = configuration.screensAndDestinations
+  }
+  
+  // MARK: - Public Methods.
+  
+  /// Sets a new `rootViewController` for the `UIWindow`.
+  /// - Parameters:
+  ///   - routable: The routable object to install as root.
+  ///   - window: The window in which to install the root.
+  ///   - completion: Optional completion to execute after making the window key and visible. Defaults to nil.
+  public func installRoot(using controller: UIViewController, in window: UIWindow, completion: (()->Void)? = nil) {
+    window.rootViewController = controller
+    window.makeKeyAndVisible()
+    completion?()
+  }
+  
+  /// Shows the routable elements in the same order they are passed in a synchronous way.
+  /// - Parameters:
+  ///   - routableElements: The screens to display with their configuration.
+  ///   - completion: An optional completion to execute after presenting all elements. Defaults to nil.
+  public func show(
+    routableElements: [AnyRoutableObject],
+    completion: RoutingCompletion? = nil
+  ) {
+    if routableElements.isEmpty {
+      return
+    }
```

Comment:
Maybe it's better with guard

Reply:
ni.. I prefer the natural logic flow. That is better `if isEmpty` then `guard !isEmpty`